tests/functional-tests/driver_test.py

```diff
@@ -2666,3 +2666,134 @@ def test_driver_one_pvc_two_pod_fail_4():
+                 ]
+    value_sc = {"volBackendFs": data["primaryFs"], "clusterId": data["id"], "permissions": "777",
+                "gid": data["gid_number"], "uid": data["uid_number"]}
+    driver_object.test_dynamic(value_sc, value_pod_passed=value_pod)
+
+
+def test_driver_sc_permissions_777_dependent_pass_1():
+    value_pod = [{"mount_path": "/usr/share/nginx/html/scale", "read_only": "False", "sub_path": ["sub_path_mnt"], "sub_path_read_only":[ False ]},
+                 {"mount_path": "/usr/share/nginx/html/scale", "read_only": "False", "sub_path": ["sub_path_mnt"], "sub_path_read_only":[ True ],
+                  "reason": "Read-only file system"},
+                 {"mount_path": "/usr/share/nginx/html/scale", "read_only": "True",
+                  "sub_path": ["sub_path_mnt"], "sub_path_read_only":[ False ], "reason": "Read-only file system"}
+                 ]
+    value_sc = {"volBackendFs": data["primaryFs"], "clusterId": data["id"], "filesetType": "dependent", "permissions": "777",
+                "gid": data["gid_number"], "uid": data["uid_number"]}
+    driver_object.test_dynamic(value_sc, value_pod_passed=value_pod)
+
+
+def test_driver_sc_permissions_666_dependent_pass_2():
+    value_pod = [{"mount_path": "/usr/share/nginx/html/scale", "read_only": "False", 
+                  "sub_path": ["sub_path_mnt"], "sub_path_read_only":[ False ], "reason": "Permission denied"},
+                 {"mount_path": "/usr/share/nginx/html/scale",
+                  "sub_path": ["sub_path_mnt"], "sub_path_read_only":[ False ],
+                  "read_only": "True", "reason": "Read-only file system"}
+                 ]
+    value_sc = {"volBackendFs": data["primaryFs"], "clusterId": data["id"], "filesetType": "dependent", "permissions": "666",
+                "gid": data["gid_number"], "uid": data["uid_number"]}
+    driver_object.test_dynamic(value_sc, value_pod_passed=value_pod)
+
+
+def test_driver_sc_permissions_777_dependent_pass_3():
+    value_pod = [{"mount_path": "/usr/share/nginx/html/scale", "read_only": "False",
+                  "sub_path": ["sub_path_mnt", "sub_path_mnt_2", "sub_path_mnt3"], "sub_path_read_only":[ False , True, True]},
+                 {"mount_path": "/usr/share/nginx/html/scale", "read_only": "False",
+                  "sub_path": ["sub_path_mnt", "sub_path_mnt_2", "sub_path_mnt3"], "sub_path_read_only":[ True, False, False], "reason": "Read-only file system"},
+                 {"mount_path": "/usr/share/nginx/html/scale", "read_only": "True",
+                  "sub_path": ["sub_path_mnt", "sub_path_mnt_2", "sub_path_mnt3"], "sub_path_read_only":[ False, False, False], "reason": "Read-only file system"}
+                 ]
+    value_sc = {"volBackendFs": data["primaryFs"], "clusterId": data["id"], "filesetType": "dependent", "permissions": "777",

```

Comment:
should we have one test for only valid  permission value without gid & uid parameter in the storageclass?